<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
	"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">

<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
	<meta content="text/html; charset=utf-8" http-equiv="Content-Type" />

	<title>: </title>

	<link type="text/css" media="screen" href="./rdoc.css" rel="stylesheet" />

</head>
<body class="file">
	
	<div id="metadata">
		<dl>
			<dt class="modified-date">Last Modified</dt>
			<dd class="modified-date">Wed Jul 20 17:59:29 +0100 2011</dd>
			
			
			
			
		</dl>

		<div class="description">
			<h2>Description</h2>
			<h1>Webservice Client Library for InterMine Data-Warehouses</h1>
<p>
This library provides an interface to the InterMine webservices API. It
makes construction and execution of queries more straightforward, safe and
convenient, and allows for results to be used directly in Ruby code. As
well as traditional row based access, the library provides an
object-orientated record result format (similar to ActiveRecords), and
allows for fast, memory efficient iteration of result sets.
</p>
<h2>Example</h2>
<p>
Get all protein domains associated with a set of genes and print their
names:
</p>
<pre>
    require &quot;intermine/service&quot;

    Service.new(&quot;www.flymine.org/query&quot;).
        new_query(&quot;Pathway&quot;)
        select(:name).
        where(&quot;genes.symbol&quot; =&gt; [&quot;zen&quot;, &quot;hox&quot;, &quot;h&quot;, &quot;bib&quot;]).
        each_row { |row| puts row[:name]}
</pre>
<h2>Who is this for?</h2>
<p>
InterMine data warehouses are typically constructed to hold Biological
data, and as this library facilitates programmatic access to these data,
this install is primarily aimed at bioinformaticians. In particular, users
of the following services may find it especially useful:
</p>
<pre>
 * FlyMine (http://www.flymine.org/query)
 * YeastMine (http://yeastmine.yeastgenome.org/yeastmine)
 * RatMine (http://ratmine.mcw.edu/ratmine)
 * modMine (http://intermine.modencode.org/release-23)
 * metabolicMine (http://www.metabolicmine.org/beta)
</pre>
<h2>How to use this library:</h2>
<p>
We have tried to construct an interface to this library that does not
require you to learn an entirely new set of concepts. As such, as well as
the underlying methods that are common to all libraries, there is an
additional set of aliases and sugar methods that emulate the DSL style of
SQL:
</p>
<h3>SQL style</h3>
<pre>
  service = Service.new(&quot;www.flymine.org/query&quot;)
  service.model.
     table(&quot;Gene&quot;).
     select(&quot;*&quot;, &quot;pathways.*&quot;).
     where(:symbol =&gt; &quot;zen&quot;).
     order_by(:symbol).
     outerjoin(:pathways).
     each_row do |r|
       puts r
     end
</pre>
<h3>Common InterMine interface</h3>
<pre>
  service = Service.new(&quot;www.flymine.org/query&quot;)
  query = service.new_query(&quot;Gene&quot;)
  query.add_views(&quot;*&quot;, &quot;pathways.*&quot;)
  query.add_constraint(&quot;symbol&quot;, &quot;=&quot;, &quot;zen&quot;)
  query.add_sort_order(:symbol)
  query.add_join(:pathways)
  query.each_row do |r|
    puts r
  end
</pre>
<p>
For more details, see the accompanying documentation and the unit tests for
interface examples. Further documentation is available at <a
href="http://www.intermine.org">www.intermine.org</a>.
</p>
<h2>Support</h2>
<p>
Support is available on our development mailing list: dev@intermine.org
</p>
<h2>License</h2>
<p>
This code is Open Source under the LGPL. Source code for all InterMine code
can be checked out from svn://subversion.flymine.org/flymine
</p>

		</div>
	</div>

	<div id="validator-badges">
	  <p><small><a href="http://validator.w3.org/check/referer">[Validate]</a></small></p>
	  <p><small>Generated with the <a href="http://deveiate.org/projects/Darkfish-Rdoc/">Darkfish 
		Rdoc Generator</a></small>.</p>
	</div>
</body>
</html>

